### PR TITLE
TimeoutF To Runner

### PR DESCRIPTION
**Pull request by nivBlox**
Move register timeout to runner and inject timeoutF from validator

### CODE REVIEW

**Comment by amir-blox** [comment]
What about spec changes? do we want to differ or patch these changes into spec?

**Comment by nivBlox** [comment]
> 

differ, as those are implamantion concern

File: protocol/v2/ssv/validator/validator.go [hunk 1]
```diff
@@ -65,9 +58,19 @@ func NewValidator(pctx context.Context, options Options) *Validator {
...
 		Signer:      options.Signer,
-		Queues:      queues,
+		Queues:      make(map[spectypes.BeaconRole]msgqueue.MsgQueue), // populate below
 		state:       uint32(NotStarted),
 	}
+
+	indexers := msgqueue.WithIndexers(msgqueue.SignedMsgIndexer(), msgqueue.DecidedMsgIndexer(), msgqueue.SignedPostConsensusMsgIndexer(), msgqueue.EventMsgMsgIndexer())
+	for _, dutyRunner := range options.DutyRunners {
+		// set timeout F
+		dutyRunner.SetTimeoutF(v.onTimeout)
```
Comment (lior-blox) [inline protocol/v2/ssv/validator/validator.go:68]: @nivBlox Lets move the SetTimoutF to base runner rather then on each

Comment (nivBlox) [reply protocol/v2/ssv/validator/validator.go]: done

**Review by amir-blox — APPROVED** [review]

**Pull request merged**